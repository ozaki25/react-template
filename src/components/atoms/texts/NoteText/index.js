import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Color from 'src/constants/Color';

const NoteText = ({ children, emphasis, red }) => (
  <Text emphasis={emphasis} red={red}>
    {children}
  </Text>
);

const Text = styled.p`
  color: ${({ red }) => (red ? Color.brand : Color.text)};
  font-size: ${({ emphasis }) => (emphasis ? '16px' : '14px')};
  font-weight: ${({ emphasis }) => (emphasis ? 'bold' : 'nomal')};
`;

NoteText.propTypes = {
  children: PropTypes.node.isRequired,
  emphasis: PropTypes.bool,
  red: PropTypes.bool,
};

NoteText.defaultProps = {
  emphasis: false,
  red: false,
};

export default NoteText;
