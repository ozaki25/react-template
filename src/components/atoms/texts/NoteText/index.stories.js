import React from 'react';

import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { text } from '@storybook/addon-knobs';

import NoteText from '.';

const stories = storiesOf('atoms/texts/NoteText', module);

stories.add(
  'nomal',
  withInfo('注意書き用文字列')(() => <NoteText>{text('text', '注意書き用文字列')}</NoteText>),
);

stories.add(
  'emphasis',
  withInfo('強調する注意書き用文字列')(() => (
    <NoteText emphasis>{text('text', '強調する注意書き用文字列')}</NoteText>
  )),
);

stories.add(
  'red',
  withInfo('赤い注意書き用文字列')(() => (
    <NoteText red>{text('text', '赤い注意書き用文字列')}</NoteText>
  )),
);

stories.add(
  'red-emphasis',
  withInfo('強調する赤い注意書き用文字列')(() => (
    <NoteText emphasis red>
      {text('text', '強調する赤い注意書き用文字列')}
    </NoteText>
  )),
);
