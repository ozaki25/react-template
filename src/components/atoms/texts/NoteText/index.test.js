import React from 'react';
import { snapshot } from 'test/helpers';
import NoteText from '.';

snapshot('NoteText/nomal', <NoteText>テキスト</NoteText>);
snapshot('NoteText/emphasis', <NoteText emphasis>テキスト</NoteText>);
snapshot('NoteText/red', <NoteText red>テキスト</NoteText>);
snapshot(
  'NoteText/red-emphasis',
  <NoteText red emphasis>
    テキスト
  </NoteText>,
);
