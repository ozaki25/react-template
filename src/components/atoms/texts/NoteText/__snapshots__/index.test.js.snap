// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: NoteText/emphasis 1`] = `
.c0 {
  color: #333;
  font-size: 16px;
  font-weight: bold;
}

<NoteText
  emphasis={true}
  red={false}
>
  <styled.p
    emphasis={true}
    red={false}
  >
    <p
      className="c0"
    >
      テキスト
    </p>
  </styled.p>
</NoteText>
`;

exports[`should match snapshot: NoteText/nomal 1`] = `
.c0 {
  color: #333;
  font-size: 14px;
  font-weight: nomal;
}

<NoteText
  emphasis={false}
  red={false}
>
  <styled.p
    emphasis={false}
    red={false}
  >
    <p
      className="c0"
    >
      テキスト
    </p>
  </styled.p>
</NoteText>
`;

exports[`should match snapshot: NoteText/red 1`] = `
.c0 {
  color: #e60000;
  font-size: 14px;
  font-weight: nomal;
}

<NoteText
  emphasis={false}
  red={true}
>
  <styled.p
    emphasis={false}
    red={true}
  >
    <p
      className="c0"
    >
      テキスト
    </p>
  </styled.p>
</NoteText>
`;

exports[`should match snapshot: NoteText/red-emphasis 1`] = `
.c0 {
  color: #e60000;
  font-size: 16px;
  font-weight: bold;
}

<NoteText
  emphasis={true}
  red={true}
>
  <styled.p
    emphasis={true}
    red={true}
  >
    <p
      className="c0"
    >
      テキスト
    </p>
  </styled.p>
</NoteText>
`;
